Automattic/newspack-newsletters
# feat: subscription lists and subscription block

### All Submissions:






### Changes proposed in this Pull Request:

Implements subscription lists management and ESP-agnostic subscription block. It currently supports Campaign Monitor, Mailchimp, and ActiveCampaign. Constant Contact is currently unavailable due to #802.

## Subscription Lists

Once a provider is selected, the existing lists become available to configure as subscription lists with a custom title and description:

### Engagement Wizard

<img width="1099" alt="image" src="https://user-images.githubusercontent.com/820752/176779916-cabdd35f-d258-49e8-9a0a-845106f6f2ae.png">

### Newsletters Settings Page

<img width="650" alt="image" src="https://user-images.githubusercontent.com/820752/176779860-4e287e1b-a78a-4d5c-8e95-7beaada2d66c.png">

## Subscription Block

Once the subscription lists are configured by an admin, an editor can use the **Newsletters Subscription Block** to configure one or multiple lists to subscribe to.

### Multiple lists with narrow width

<img width="426" alt="image" src="https://user-images.githubusercontent.com/820752/176778030-5130b1d3-6b9d-4e76-8db4-bcfb801312bd.png">

### Multiple lists with wider width

<img width="847" alt="image" src="https://user-images.githubusercontent.com/820752/176778179-bff5784f-9640-4a0c-a24d-18ebbda25392.png">

### Single list

<img width="853" alt="image" src="https://user-images.githubusercontent.com/820752/176778233-2b298d6d-56b2-45e1-92f5-3ba7ae8ee80f.png">



Closes #430
Closes #636 

### How to test the changes in this Pull Request:

1. Check out this branch and https://github.com/Automattic/newspack-plugin/pull/1734
2. Visit the Engagement wizard, select a provider you have multiple lists and save
3. Confirm the "Subscription Lists" section is updated once the config is saved
4. Enable, apply custom titles and descriptions to your lists and save
5. Edit a page and add a "Newsletter Subscription Form" block
6. Tweak different values and list selections and confirm they behave as expected
7. Save the page with the block with multiple lists selected
8. Subscribe an email selecting some but not all lists
9. Confirm on the ESP that the email is subscribed
10. Test subscription with all 3 supported ESPs
11. Access the page with mobile viewport and confirm the lists are displayed vertically

### Other information:
